test/agent.test.js

```diff
@@ -1148,6 +1148,21 @@ test('#flush()', function (t) {
     })
   })
 
+  t.test('flush can be used without a callback to return a Promise', function (t) {
+    t.plan(1)
+
+    const agent = new Agent()
+
+    agent.flush().then(function () {
+      t.pass('should resolve the Promise for agent.flush')
+    }).catch(function (err) {
+      t.error(err, 'no error passed to agent.flush callback')
+    }).finally(function () {
+      agent.destroy()
+      t.end()
+    })
```

Comment:
Unfortunately we still support and test with node v8, which [doesn't support Promise.finally](https://node.green/#ES2018-features-Promise-prototype-finally). I think it would be fine to put the `agent.destroy(); t.end()` in the `.then(...)` handler and call it a day.

Reply:
Ah yes, those were the days! Agreed, tweaked this in 6a6afea.